lib/vagrant-cachier/bucket/mvn.rb

```diff
@@ -0,0 +1,21 @@
+module VagrantPlugins
+  module Cachier
+    class Bucket
+      class Mvn < Bucket
+        def self.capability
+          :maven_cache_dir
+        end
+
+        def install
+          if guest.capability?(:maven_cache_dir)
+            guest_path = guest.capability(:maven_cache_dir)
+            return if @env[:cache_dirs].include?(guest_path)
+            symlink(guest_path)
```

Comment:
You have to do something like 
```
symlink(guest_path) if guest_path
``` 
Otherwise your check for maven will have no effect.